src/coreclr/jit/compiler.cpp

```diff
@@ -5834,12 +5834,18 @@ void Compiler::generatePatchpointInfo()
     //
     const int totalFrameSize = codeGen->genTotalFrameSize() + TARGET_POINTER_SIZE;
     const int offsetAdjust   = 0;
-#elif defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+#elif defined(TARGET_ARM64)
     // SP is not manipulated by calls so no frame size adjustment needed.
     // Local Offsets may need adjusting, if FP is at bottom of frame.
     //
     const int totalFrameSize = codeGen->genTotalFrameSize();
     const int offsetAdjust   = codeGen->genSPtoFPdelta() - totalFrameSize;
+#elif defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+    // SP is not manipulated by calls so no frame size adjustment needed.
+    // Local Offsets don't need adjusting for LoongArch64.
+    //
+    const int totalFrameSize = codeGen->genTotalFrameSize();
+    const int offsetAdjust   = 0;
```

Comment:
I don't understand why we need to add a field to the patchpoint information. Can't we just change the `offsetAdjust` here such that the offsets are encoded relative to the bottom of the frame like before this change? That should give us all the necessary information to compute the offsets into the tier 0 frame.

Reply:
The patchpoint information should save the whole tier0 frame’s size and the tier0's FP relative to SP.
They are all needed by tier1.

Reply:
What is tier0 FP relative to SP needed for in the tier1 frame?

Reply:
> What is tier0 FP relative to SP needed for in the tier1 frame?

The locals which are shared with tier1 frame, their tier0's offsets should be updated to relative tier1's FP.

Reply:
> What I am suggesting here is to encode their offsets in the patch point information in tier-0 so that this is unnecessary. Then the field isn't needed in tier-1 anymore.

What you mean is `offsetAdjust = codeGen->genSPtoFPdelta()`, and then avoid adding the `SetCalleeSaveRegistersOffset()`

Reply:
Yes, exactly.

Reply:
That's a good idea.
Thanks very much.